druid/src/lens.rs

```diff
@@ -91,6 +93,59 @@ pub trait LensExt<A: ?Sized, B: ?Sized>: Lens<A, B> {
     {
         Then::new(self, other)
     }
+
+    /// Invoke a type's `Deref` impl
+    ///
+    /// ```
+    /// # use druid::*;
+    /// let lens = lens!((bool, Box<u32>), 1).deref();
+    /// assert_eq!(lens.get(&(false, Box::new(42))), 42);
+    /// ```
+    fn deref(self) -> Then<Self, Deref, B>
+    where
+        B: ops::Deref + ops::DerefMut,
+        Self: Sized,
+    {
+        self.then(Deref)
+    }
+
+    /// Access an index in a container
+    ///
+    /// ```
+    /// # use druid::*;
+    /// let lens = lens!((bool, Vec<u8>), 1).deref().index(2);
+    /// assert_eq!(lens.get(&(false, vec![0, 1, 2, 3])), 2);
+    /// ```
+    fn index<I>(self, index: I) -> Then<Self, Index<I>, B>
+    where
+        I: Clone,
+        B: ops::Index<I> + ops::IndexMut<I>,
+        Self: Sized,
+    {
+        self.then(Index::new(index))
+    }
+
+    /// Adapt to operate on the contents of an `Arc` with efficient copy-on-write semantics
+    ///
+    /// ```
+    /// # use druid::*; use std::sync::Arc;
+    /// let lens = lens::Id.index(2).in_arc();
+    /// let mut x = Arc::new(vec![0, 1, 2, 3]);
+    /// let original = x.clone();
+    /// assert_eq!(lens.get(&x), 2);
+    /// lens.put(&mut x, 2);
+    /// assert!(Arc::ptr_eq(&original, &x), "no-op writes don't cause a deep copy");
+    /// lens.put(&mut x, 42);
+    /// assert_eq!(&*x, &[0, 1, 42, 3]);
+    /// ```
+    fn in_arc(self) -> InArc<Self>
```

Comment:
Bikeshedding still welcome, though I think this is probably adequate.

Reply:
I can't think of anything better? maybe just `arc`, but either or.